src/menu.rs

```diff
@@ -155,15 +161,21 @@ pub fn run(mut window: &mut PistonWindow,
                             match menu_selection {
                                 MenuSelection::Play => {
                                     music::play(&Music::Action, music::Repeat::Forever);
-                                    game::Game::new(window_size)
-                                        .run(&mut window, &mut opengl, &mut glyph_cache);
+                                    game::Game::new(window_size).run(&mut events,
+                                                                     &mut window,
+                                                                     &mut opengl,
+                                                                     &mut glyph_cache);
                                     music::play(&Music::Menu, music::Repeat::Forever);
                                 }
                                 MenuSelection::Story => {
-                                    story::run(&mut window, &mut opengl, &mut glyph_cache);
+                                    story::run(&mut events,
```

Comment:
I was able to create a trait which adds the `.next()` method to `GlutinWindow`, but it's kind of ugly. You can check it out on my branch `glutin-window-next` [here](https://github.com/bsvercl/rust-belt/tree/glutin-window-next).